File: app/src/main/java/com/sakuna63/tumbin/application/contract/LoginContract.java
package com.sakuna63.tumbin.application.contract;

import android.support.annotation.NonNull;

public interface LoginContract {
    interface View extends BaseView<Presenter> {

        void setLoginButtonActive(boolean active);

        void setLoginProgress(boolean visible, @NonNull String message);

        void showErrorMessage(@NonNull String message);

        void navigateToHome();

        void navigateToLoginPage(@NonNull String url);
    }

    interface Presenter extends BasePresenter {
        void onClickLogin();

        void onLoginCallback(@NonNull String dataString);
    }
}

File: app/src/main/java/com/sakuna63/tumbin/application/contract/LoginContract.kt
package com.sakuna63.tumbin.application.contract

interface LoginContract {
    interface View : BaseView<Presenter> {

        fun setLoginButtonActive(active: Boolean)

        fun setLoginProgress(visible: Boolean, message: String)

        fun showErrorMessage(message: String)

        fun navigateToHome()

        fun navigateToLoginPage(url: String)
    }

    interface Presenter : BasePresenter {
        fun onClickLogin()

        fun onLoginCallback(dataString: String?)
    }
}
